### PR TITLE
fix: NetManager conversions and removal of WebManager

### PR DESCRIPTION
**Pull request by magicus**
This is a PR based on top of https://github.com/Wynntils/Artemis/pull/825. (This targets the `netmanager` branch, not `main`)

I have tried to do just one simple step per commit, and explain what I do. Most are straightforward conversions, some require a bit extra job (but the commit comment explains that).

When both #825 and this has been included in main, the "webapi rewrite" is done.

### CODE REVIEW

**Comment by magicus** [comment]
I chose `fix:` instead of `refactor:`, since some actual bug fixes are included.

File: common/src/main/java/com/wynntils/commands/ServerCommand.java
Comment (kristofbolyai) [inline common/src/main/java/com/wynntils/commands/ServerCommand.java:85]: I am pretty sure this works, but it is a bit weird that we send a failure, but then, send a "success" as well.

File: common/src/main/java/com/wynntils/wynn/model/SplashManager.java
Comment (kristofbolyai) [inline common/src/main/java/com/wynntils/wynn/model/SplashManager.java:36]: I don't think such "simple" types cannot be handled by GSON without a Type, but this should do.

File: common/src/main/java/com/wynntils/wynn/model/map/MapModel.java
Comment (kristofbolyai) [inline common/src/main/java/com/wynntils/wynn/model/map/MapModel.java]: Was this really an IOException? Can we use a different logging message here, so we know where we failed? Also, perhaps log the error itself?

File: common/src/main/java/com/wynntils/wynn/model/map/MapModel.java
Comment (kristofbolyai) [inline common/src/main/java/com/wynntils/wynn/model/map/MapModel.java]: Is it worth to log the exception here?

Comment (kristofbolyai) [reply common/src/main/java/com/wynntils/wynn/model/map/MapModel.java]: (I mean logging IOException e)

File: common/src/main/java/com/wynntils/core/net/athena/WynntilsAccountManager.java
Comment (kristofbolyai) [inline common/src/main/java/com/wynntils/core/net/athena/WynntilsAccountManager.java:116]: Didn't the commit description just say this is now up to date? Or what is this about?

Comment (kristofbolyai) [reply common/src/main/java/com/wynntils/core/net/athena/WynntilsAccountManager.java:116]: Is this about the users of the token does not respect loading this is async, hence making the code weird?

File: common/src/main/java/com/wynntils/commands/WynntilsCommand.java
Comment (kristofbolyai) [inline common/src/main/java/com/wynntils/commands/WynntilsCommand.java]: Shouldn't we reset the other managers? (I know this is bad as a whole, so maybe we can just ignore this for the time being)

**Review by kristofbolyai — COMMENTED** [review]
I have checked every commit now. I've left comments everywhere related. Once those are resolved, I am happy to approve this.

File: common/src/main/java/com/wynntils/commands/ServerCommand.java
Comment (magicus) [reply common/src/main/java/com/wynntils/commands/ServerCommand.java:85]: Yes, it's kind of weird. Sounds almost as weird to call `sendSuccess` as well. I looked at the implementation, both methods do more or less the same. 

This was not really part of the real fix, just something I noticed while rummaging about in the ServerCommand source and thought was an, unrelated, improvement. I can leave it out for now as well, too, and we can revisit this separately and see if there is some better way (maybe a third `sendUpdate` method?) to do this.

Comment (kristofbolyai) [reply common/src/main/java/com/wynntils/commands/ServerCommand.java:85]: This is fine. No need to revert it.

File: common/src/main/java/com/wynntils/wynn/model/SplashManager.java
Comment (magicus) [reply common/src/main/java/com/wynntils/wynn/model/SplashManager.java:36]: What do you mean? That I should use the GSON.fromJson(reader, Class<?> class) version instead? It clearly says "It is not suitable to use if the specified class is a generic type since it will not have the generic type information because of the Type Erasure feature of Java. Therefore, this method should not be used if the desired type is a generic type. "

Comment (kristofbolyai) [reply common/src/main/java/com/wynntils/wynn/model/SplashManager.java:36]: I remember a GSON method without a Type/Class. Anyways, I think I miss remember. I've been debugging matrixes for an hour before writing this 🙃.

File: common/src/main/java/com/wynntils/core/net/athena/WynntilsAccountManager.java
Comment (magicus) [reply common/src/main/java/com/wynntils/core/net/athena/WynntilsAccountManager.java:116]: > Is this about the users of the token does not respect loading this is async, hence making the code weird?

Yes, exactly. Code like Hades is assuming that when the HadesManager or whatever it was called is being init'ed, that the login is done so we have a token. If not, it complains loudly and refuses to make any further attempts at connecting.

I think we should rewrite this so that we send out like a AthenaLoginEvent when we are authenticated with Athena, and that in turn could trigger Hades to log in, etc.

But that was not something I wanted to address here.

File: common/src/main/java/com/wynntils/commands/WynntilsCommand.java
Comment (magicus) [reply common/src/main/java/com/wynntils/commands/WynntilsCommand.java]: So, to be clear, I have copied as good as humanely possible what the old WebManager.reset() did. It held on to a few, but not all, of the "profile" data. These are the ones I have called init on.

Comment (magicus) [reply common/src/main/java/com/wynntils/commands/WynntilsCommand.java]: I thought I was clear enough that the entire thing was a mess that needs to be addressed by https://github.com/Wynntils/Artemis/issues/824 :)

File: common/src/main/java/com/wynntils/wynn/model/SplashManager.java
Comment (magicus) [reply common/src/main/java/com/wynntils/wynn/model/SplashManager.java:36]: No, there is not. But if you don't have generics, you can do like `GSON.fromJson(reader, FluffyProfile.class)`.

File: common/src/main/java/com/wynntils/commands/WynntilsCommand.java
Comment (kristofbolyai) [reply common/src/main/java/com/wynntils/commands/WynntilsCommand.java]: Yes, this is bad. Just one question. Do you have plans to address this yourself, or should I be looking into it sometime?

File: common/src/main/java/com/wynntils/core/net/athena/WynntilsAccountManager.java
Comment (kristofbolyai) [reply common/src/main/java/com/wynntils/core/net/athena/WynntilsAccountManager.java:116]: Agreed. I think we should have an issue for this though.

Comment (kristofbolyai) [reply common/src/main/java/com/wynntils/core/net/athena/WynntilsAccountManager.java:116]: https://github.com/Wynntils/Artemis/issues/829

File: common/src/main/java/com/wynntils/commands/WynntilsCommand.java
Comment (magicus) [reply common/src/main/java/com/wynntils/commands/WynntilsCommand.java]: I can look into it. I think the entire code base could do with another sweep, which looks more closely at error handling and exceptional conditions.

However.

I'm currently quite fed up with this, I've been working on this for two weeks (not counting the time I was contemplating the design prior to that) and I'd like to take a break. So if you think it needs to be fixed "soon-ish", maybe you should to it.

Comment (kristofbolyai) [reply common/src/main/java/com/wynntils/commands/WynntilsCommand.java]: Okay. That is understandable. We will see who gets to this first. I will let you know, if I start working on this (and it would be great if you could do this for me too).

Comment (magicus) [reply common/src/main/java/com/wynntils/commands/WynntilsCommand.java]: Sure, I'll let you know. (In general, it is probably a good idea to assign a bug to yourself, and also perhaps add a comment, if you intend to start solving it)

**Review by kristofbolyai — APPROVED** [review]
Looks great to me! I'm excited about this. (I would say this PR does not need explicit approval from someone else, as it is quite trivial, and is just about porting to use the new handlers). But you decide.

**Comment by magicus** [comment]
I'm also eager to have this go into main, so I think I'm satisfied with your approval only.

**Pull request merged**
